engine/runtime-compiler/src/main/java/org/enso/compiler/data/BindingsMapBase.java

```diff
@@ -0,0 +1,133 @@
+package org.enso.compiler.data;
+
+import static org.enso.scala.wrapper.ScalaConversions.nil;
+
+import java.util.function.Function;
+import java.util.function.Supplier;
+import org.enso.compiler.data.BindingsMap.DefinedEntity;
+import org.enso.compiler.data.BindingsMap.ModuleReference;
+import org.enso.compiler.data.BindingsMap.ResolvedImport;
+import org.enso.compiler.data.BindingsMap.ResolvedName;
+import org.enso.compiler.pass.IRPass;
+import scala.collection.immutable.List;
+import scala.collection.immutable.Map;
+import scala.collection.immutable.Map$;
+
+/**
+ * A utility structure for resolving symbols in a given module. Represents immutable (as much as
+ * possible) view of a "binding map". The {@code state} can mutate, but it is guaranteed to be
+ * changed atomically to ensure consistency - all elements of the state are always changed at once.
```

Comment:
Would be nice to add this link to javadoc:
```
* @See <a href="http://wiki.apidesign.org/wiki/TransactionalDataStructure">Transactional data structures</a>
```

Reply:
- I'd love to spread links to my website!
- however the link may not last as long as the Javadoc in the immutable git history
- thus I'd restrain myself and only add the links into PR comments that can change in the future
- great description of "transactional data structures" is [available here](http://wiki.apidesign.org/wiki/TransactionalDataStructure)